build_manpages/compat.py

```diff
@@ -1,48 +1,28 @@
 """
-Compatibility hacks for the argparse-manpage project.
+Compat hacks for argparse-manpage's build_manpages module
 """
 
-import sys
+# pylint: disable=unused-import,deprecated-module,raise-missing-from
 
-# Drop once Python 2.7 is dropped
-# pylint: disable=unused-import
 try:
-    from configparser import ConfigParser
+    from setuptools import Command
+    from setuptools.command.build_py import build_py
+    from setuptools.command.install import install
 except ImportError:
-    from ConfigParser import SafeConfigParser as ConfigParser  # type: ignore
+    try:
+        from distutils.core import Command
+        from distutils.command.build_py import build_py
+        from distutils.command.install import install
+    except ImportError as orig_err:
+        raise ImportError(
+            "To use the 'build_manpages' tool on Python 3.12+, "
+            "you need to install 'setuptools'."
+        )
```

Comment:
It makes sense, but to make it better for the dependents, I'd still prefer to have an extra dependency.

Reply:
Ok, I added:
```
    extras_require={
        "build-manpages": ['setuptools;python_version>="3.12"'],
    },
```
I have never done this before, so hopefully, that helps with pip installations.  For the RPM, the dynamic build-requires generator doesn't seem to use this metadata.

Reply:
> For the RPM, the dynamic build-requires generator doesn't seem to use this metadata.

It does, you just need to package the extra manually. https://docs.fedoraproject.org/en-US/packaging-guidelines/Python/#Extras

Then, when packagers use `argparse-manpage[build-manpages]` as a dependency upstream it'll work in RPM.
You might want to change the exception message to say `argparse-manpage[build-manpages]` instead of `setuptools`. You might want to mention this in the README.

Reply:
Thanks for all the info. I'll make it a hard dep for now with a TODO item (`extras_require` statement complicates builds for EL7/Python 2.7; EL7/Python 2.7 support could be dropped, but I'd like to keep it for another version).